pkg/geo/summary.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geo
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/geo/geopb"
+	"github.com/cockroachdb/errors"
+	"github.com/twpayne/go-geom"
+)
+
+// Summary returns a text summary of the contents of the geometry type.
+//
+// Flags shown square brackets after the geometry type have the following meaning:
+// M: has M coordinate
+// Z: has Z coordinate
+// B: has a cached bounding box
+// G: is geography
+// S: has spatial reference system
+func Summary(t geom.T, isGeography bool, offset int) (summary string, err error) {
```

Comment:
nit: taking `offset` as an initial argument for the public facing function makes the interface a little unfriendly as there is a dangling magic constant `0`.

Can we do something like:
```
func Summary(t geom.T, isGeography bool) (summary string, err error) {
   return summary(t, isGeography, 0)
}

func summary(t geom.T, isGeography bool, spaceOffset int) (summary string, err error) {
}
```
```

Reply:
Done.